src/Mvc/Mvc.Core/src/Infrastructure/LocalRedirectResultExecutor.cs

```diff
@@ -45,36 +44,52 @@ public LocalRedirectResultExecutor(ILoggerFactory loggerFactory, IUrlHelperFacto
         /// <inheritdoc />
         public virtual Task ExecuteAsync(ActionContext context, LocalRedirectResult result)
         {
-            if (context == null)
+            var urlHelper = result.UrlHelper ?? _urlHelperFactory.GetUrlHelper(context);
+
+            return ExecuteAsyncInternal(
+                context.HttpContext,
+                result,
+                urlHelper.IsLocalUrl,
```

Comment:
I feel a little iffy about this. This is now going to allocate a delegate per invocation. Could we have it fall-back in the implementation instead e.g.

```C#
internal static Task ExecuteAsyncInternal(
            HttpContext httpContext,
            LocalRedirectResult result,
            IUrlHelper? urlHelper)
{
  ...
   var isLocalUrl = urlHelper?.IsLocalUrl(url) ?? UrlHelperBase.CheckIsLocalUrl(url);
}